wordpress-mobile/WordPress-iOS
# Media Library: Add basic upload functionality

This PR adds upload functionality to the media library.

![simulator screen shot 23 may 2017 23 26 36](https://cloud.githubusercontent.com/assets/4780/26379159/a02022da-400f-11e7-8e61-e6742e62ee1d.png)

You can upload media either by tapping the new + button in the nav bar, or by tapping the Upload Media button in the no results view if you have no media.

During upload, we're currently showing blocking progress using the HUD. I think we can ditch this fairly soon, but I wanted to go for the simplest implementation first. You can tap on the HUD to cancel any pending uploads.

To test:

* Attempt to upload media!
* Try from an empty library and a library with existing content.
* Try cancelling uploads
* Try from dotcom and self hosted sites
* Try both photos and videos
* Note that progress reporting seems to be more granular on a device vs in the simulator

Needs review: @kurzee 

@SergioEstevao I also made a couple of tweaks to the media library data source and the media service – would you mind double checking those?
